Content.Server/Anomaly/Effects/LiquidAnomalySystem.cs

```diff
@@ -0,0 +1,128 @@
+    [Dependency] private readonly PuddleSystem _puddleSystem = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<LiquidAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+        SubscribeLocalEvent<LiquidAnomalyComponent, AnomalySupercriticalEvent>(OnSupercritical);
+        SubscribeLocalEvent<LiquidAnomalyComponent, AnomalySeverityChangedEvent>(OnSeverityChanged);
+        SubscribeLocalEvent<LiquidAnomalyComponent, MapInitEvent>(OnMapInit);
+    }
+    private void OnMapInit(EntityUid uid, LiquidAnomalyComponent component, MapInitEvent args)
+    {
+        ChangeReagentType(uid, component);
+    }
+
+    private void OnPulse(EntityUid uid, LiquidAnomalyComponent component, ref AnomalyPulseEvent args)
+    {
+        PulseScalableEffect(
+            uid,
+            component.MaxSolutionGeneration * args.Severity,
+            component.MaxSolutionInjection * args.Severity,
+            component.InjectRadius,
+            component.Reagent);
+    }
+
+    private void OnSupercritical(EntityUid uid, LiquidAnomalyComponent component, ref AnomalySupercriticalEvent args)
+    {
+        PulseScalableEffect(
+            uid,
+            component.SuperCriticalSolutionGeneration,
+            component.SuperCriticalSolutionInjection,
+            component.SuperCriticalInjectRadius,
+            component.Reagent);
+    }
+
+
+    private void OnSeverityChanged(EntityUid uid, LiquidAnomalyComponent component, ref AnomalySeverityChangedEvent args)
+    {
+        //If after the severity change, its level has exceeded the threshold, the type of reagent changes, and the threshold increases.
+        if (args.Severity >= component.NextChangeThreshold)
+        {
+            component.NextChangeThreshold += component.ReagentChangeStep;
+
+            ChangeReagentType(uid, component);
+            return;
+        }
+    }
+
+    private void ChangeReagentType(EntityUid uid, LiquidAnomalyComponent component)
+    {
+        if (component.PossibleChemicals.Count == 0) return;
+
+        var rndIndex = _random.Next(0, component.PossibleChemicals.Count - 1);
+        var reagent = component.PossibleChemicals[rndIndex];
+
+        component.Reagent = reagent;
+        var color = _proto.Index<ReagentPrototype>(reagent).SubstanceColor;
+        _light.SetColor(uid, color);
+
+        //Spawn Effect
+        var uidXform = Transform(uid);
+        Spawn("PuddleSparkle", uidXform.Coordinates);
+        _audio.PlayPvs(component.ChangeSound, uid);
+    }
+
+    private void PulseScalableEffect(
+        EntityUid uid,
+        float solutionAmount,
+        float maxInject,
+        float injectRadius,
+        string reagent)
+    {
+        //We get all the entity in the radius into which the reagent will be injected.
+        var injectableQuery = GetEntityQuery<InjectableSolutionComponent>();
```

Comment:
store these on the system

Reply:
What? What for?

Reply:
to reuse them, like xformquery in other system